src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -127,13 +129,33 @@ public async Task Invoke(HttpContext context)
         // Assumes the response headers have not been sent.  If they have, still attempt to write to the body.
         private Task DisplayException(ErrorContext errorContext)
         {
-            var compilationException = errorContext.Exception as ICompilationException;
-            if (compilationException != null)
+            var httpContext = errorContext.HttpContext;
+            var acceptHeader = httpContext.Request.Headers[HeaderNames.Accept].ToString();
+
+            // If the client does not ask for HTML just format the exception as plain text
+            if (acceptHeader != null && !acceptHeader.Contains("text/html", StringComparison.OrdinalIgnoreCase))
+            {
+                httpContext.Response.ContentType = "text/plain";
+
+                var sb = new StringBuilder();
+                sb.AppendLine(errorContext.Exception.ToString());
+                sb.AppendLine();
+                sb.AppendLine("HEADERS");
+                sb.AppendLine("=======");
+                foreach (var pair in httpContext.Request.Headers)
+                {
+                    sb.AppendLine($"{pair.Key} = {pair.Value}");
```

Comment:
```suggestion
                    sb.AppendLine($"{pair.Key}: {pair.Value}");
```

Reply:
Just to be pedantic since this change is taking longer than it needs to. What's wrong with `=`?

Reply:
It's just not the expected format for HTTP headers.

Reply:
It's ok this isn't an HTTP response, it's a plaintext rendering so I'm going to leave it as `=`

Reply:
For more examples the chrome and IE dev tools also render it as `header: value`. That's the layout people are used to seeing headers in. `header = value` looks like query parameters.

Reply:
> For more examples the chrome and IE dev tools also render it as header: value. That's the layout people are used to seeing headers in.

That's a fair point.

> header = value looks like query parameters.

Good thing we have a giant HEADERS title in the respones 😄